pennylane/ops/qubit/non_parametric_ops.py

```diff
@@ -864,14 +865,51 @@ def __init__(
 
         self._target_wire = wires[0]
         self._work_wires = work_wires
+        self._control_wires = control_wires
+
+        U = PauliX.matrix
+        self.U = U
+
+        wires = control_wires + wires
+
+        if not control_values:
+            control_values = "1" * len(control_wires)
+
+        control_int = self._parse_control_values(control_wires, control_values)
+        self.control_values = control_values
+
+        self._padding_left = control_int * len(U)
```

Comment:
I bet we could get rid of the `self._padding_left` and `self.padding_right` attributes, and just calculate them within `self._matrix` temporarily when they're needed to compute `self._CX`

Reply:
Yup on it!

Reply:
@albi3ro If I am not wrong, to run all the checks in `_parse_control_values`, it should be called in `__init__`. To do this, I think `control_int` needs to be used in the method somewhere otherwise an error is coming. 
So should the `self._padding_left` and `self._padding_right` be calculated in the `__init__` itself then?